atomic_reactor/plugins/exit_sendmail.py

```diff
@@ -123,13 +124,19 @@ def __init__(self, tasker, workflow,
         else:
             self.log.info("Koji task ID: %s", self.koji_task_id)
 
-        try:
-            self.koji_build_id = self.workflow.exit_results.get(KojiPromotePlugin.key)
-        except Exception:
+        valid_id = False
+        for key in [KojiImportPlugin.key, KojiPromotePlugin.key]:
```

Comment:
I'm trying to understand why this line was even wrapped in a try-except block. I think it's because some of the unit tests for this plugin are using a mocked version of `DockerBuildWorkflow` which does not always have an `exit_results` attribute. This, of course, is not a valid scenario because `DockerBuildWorkflow` always has this attribute (https://github.com/projectatomic/atomic-reactor/blob/master/atomic_reactor/inner.py#L273).

I propose removing this try-except block. Some of the tests for this plugin may fail. To fix, simply set `exist_results = {}` in the `WF` class for the corresponding tests. Thoughts?